featuretools/primitives/standard/transform_primitive.py

```diff
@@ -253,6 +257,21 @@ class Diff(TransformPrimitive):
         >>> values = [1, 10, 3, 4, 15]
         >>> diff(values).tolist()
         [nan, 9.0, -7.0, 1.0, 11.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> values = [1, 2, 4, 7, 11, 16]
+        >>> diff_periods = Diff(periods = 1)
+        >>> diff_periods(values).tolist()
+        [nan, nan, 1.0, 2.0, 3.0, 4.0]
+
+        Using datetimes
```

Comment:
I think we'd need to add an input_type to allow datetime columns to actually be used by this primitive. So instead of `input_types = [ColumnSchema(semantic_tags={"numeric"})]` it could be `input_types = [[ColumnSchema(semantic_tags={"numeric"})], [ColumnSchema(logical_type=Datetime)]]`.

We might want to consider requiring it be a time index even, since something like diff only is relevant when there's an ordering to the data...

Reply:
Yeah I was confused by this since technically I can use a datetime series no problem. I will add that in.

Reply:
Yep the input types won't stop you from calling the primitive's function on whatever series you choose, but it will impact which columns a primitive is applied to automatically in dfs.